roles/kube_prometheus/hack/aggregate_dashboard.py

```diff
@@ -0,0 +1,194 @@
+        return self._spec
+
+    @staticmethod
+    def _is_dashboard(spec):
+        return 'annotations' in spec and 'rows' in spec
+
+    @property
+    def source(self):
+        """return source of dashboard, None if not found"""
+        return self.spec.get('__inputs', [{}])[0].get('name')
+
+    @property
+    def basename(self):
+        """return the basename of dashboard
+
+        either the name of the file or the single json key
+        that host the entire dashboard if any
+        """
+        if self._basename is None:
+            self._load_dashboard()
+        return self._basename
+
+    @property
+    def grafana_name(self):
+        """return basename(of the filename) of the dashboard"""
+        return os.path.basename(self.basename).replace(
+            '.json', '-dashboard.json')
+
+    def compute_dashboard_string(self):
+        spec = dict(self.spec)
+        spec.pop('id', None)  # Clean id
+        json_dashboard = json.dumps({
+                'inputs': [{
+                    'name': self.source or self.DEFAULT_SOURCE,
+                    'type:': 'datasource',
+                    'pluginId': 'prometheus'
+                }],
+                'overwrite': True,
+                'dashboard': spec
+            },
+            sort_keys=True,
+            indent=2,
+            separators=(',', ': ')
+        )
+        return '{name}: |+\n{json}'.format(
+            name=self.grafana_name,
+            json=indent(json_dashboard, ' '*2)
+        )
+
+
+class DashboardAggregator:
+
+    def __init__(self, configfile=None, dashboard_class=Dashboard):
+        self.configfile = configfile
+        self.dashboard_class = Dashboard
+        self._config = None
+
+    def parse_cli_args(self):
+        parser = argparse.ArgumentParser(
+            description='Aggregate json dashboard for grafana in kubernetes',
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+        parser.add_argument(
+            '--configfile', '-c',
+            default='./aggregate_dashboard.ini',
+            type=argparse.FileType('r'),
+            help='path to the configfile')
+        parser.add_argument(
+            '--force', '-f', default=False, type=bool,
+            help='force loading of dashboard even if not recognize as such')
+        args = parser.parse_args()
+        self.configfile = args.configfile
+        self.force = args.force
+
+    @property
+    def config(self):
+        if self._config is None:
+            ini_parser = configparser.ConfigParser()
+            ini_parser.readfp(self.configfile)
+            self._config = ini_parser
+        return self._config
+
+    @property
+    def confdir(self):
+        return os.path.dirname(self.configfile)
+
+    def iter_sources(self):
+        for section in self.config.sections():
+            if section.startswith('source:'):
+                yield section
+
+    def iter_filenames_in_source(self, source_name):
```

Comment:
Some nasty bug here if you have a dir call `src/\*.json` and source that target `src/*.json`. Also look at current directory not relative to the configuration file